pinot-core/src/main/java/org/apache/pinot/core/segment/processing/genericrow/GenericRowFileReader.java

```diff
@@ -77,6 +133,13 @@ public void read(int rowId, GenericRow buffer) {
    * Compares the rows at the given row ids. Only compare the values for the sort fields.
    */
   public int compare(int rowId1, int rowId2) {
+    // Sort with the extracted sort fields for better data locality.
+    if (_sortFieldsOffsetBuffer != null) {
+      long offset1 = _sortFieldsOffsetBuffer.getLong((long) rowId1 << 3); // rowId1 * Long.BYTES
+      long offset2 = _sortFieldsOffsetBuffer.getLong((long) rowId2 << 3); // rowId2 * Long.BYTES
```

Comment:
because `Long.BYTES` is a constant (it is `static final`) `* Long.BYTES` will get strength reduced to `<< 3` anyway.